### DISCUSSION TITLE
When using get_oauth_router, how is "SECRET" used?

### DISCUSSION THREAD

**Discussion by Cecron** [fastapi-users/fastapi-users#1078] [Q&A] [answered]
I followed the [OAuth2 Full SQLAlchemy example](https://fastapi-users.github.io/fastapi-users/10.1/configuration/oauth/#sqlalchemy_1) and changed the **SECRET** in `app/users.py`:

    SECRET = "SECRET"

but what about the  **SECRET** in `app/app.py`:

    fastapi_users.get_oauth_router(google_oauth_client, auth_backend, "SECRET"),

Running the example works whatever I do, but what does this **SECRET** do?
Is this a totally separate **SECRET** that should have it's own unique value?

**Comment by frankie567** [comment, marked as answer]
Hi @Cecron 👋 

Well, this is a miss in the OAuth2 example but clearly yes, this should be the same `SECRET`. It's used to sign the state token during the OAuth authentication. So, its value doesn't really matter as long as it's a strong value you keep secret (like a password).

I'll fix the example!
